mlp_tbb.cc

```diff
@@ -146,14 +176,19 @@ class FullyConnectedForward {
     double t0 = dsecnd();
     int m = input_->nrows(), n = output_->ncols(), k = input_->ncols();
 
+#if COUNT_NODES
+    ++num_node_executions;
+#endif
     // forward gemm performs well with aspect ratio
     // (m_end - m_begin) ~= 4 * (n_end - n_begin)
     float aspect_ratio = 4.;
 
     tbb::parallel_for(
         0,
-        nthreads_per_socket,
+        SP*nthreads_per_socket,
         [&](size_t task_id) {
+          double sgst = dsecnd();
+          int tid = numa_node_id_ * nthreads_per_socket + task_id;
```

Comment:
If a thread grabs more than 1 task, we want to account the execution time of the multiple tasks to the thread. This is a reason why I kept track execution time based on current_thread_index . Please let me know your thought.